### DISCUSSION TITLE
Support for non-Latin scripts

### DISCUSSION THREAD

**Discussion by JacobMartinsen** [kantord/LibreLingo#2873] [Q&A]
Hello everyone. I am an developer working with a Western Syriac/Aramaic teacher pro-bono to create materials to teach the language to new students, and we were exploring your application to help us create an easy way to introduce and teach to a new generation of inquirers that expect modern way of learning. The only hurdle seems to be the script, which has an official [Unicode script](https://en.wikipedia.org/wiki/Syriac_alphabet#Unicode), and some [community-created scripts](https://sedra.bethmardutho.org/about/fonts)

The major scripts used for Western Syriac today are Serto, a script that lends itself to quick handwriting, and Estrangelo, a more formal, cursive script; both are non-latin.  

I see that there is a project related to this. Is the issue around core support of non-Latin scripts like displaying non-Latin characters on the UI or in the course creation scripts, or are the remaining issues related to facilitating the use of non-Latin scripts, like having an on-screen keyboard? Would we be safe if we started creating a course using the documented method with a non-Latin script for Syriac? 

Thanks!

**Comment by kantord** [comment]
There are 3 main issues:

1. We need a virtual keyboard to support typing in a different script as a lot of users might have trouble using multiple different keyboard layouts. There is an issue for this: https://github.com/LibreLingo/LibreLingo/issues/1129
2. The students would need to learn the other script, so some way the app should be able to teach it. There are issues for that: https://github.com/LibreLingo/LibreLingo/issues/1143 https://github.com/LibreLingo/LibreLingo/issues/1072 
3. Poor RTL support at the moment: https://github.com/LibreLingo/LibreLingo/issues/1837
